src/main/scala/com/ubirch/webui/models/keycloak/util/BareKeycloakUtil.scala

```diff
@@ -705,10 +705,12 @@ case class MemberResourceRepresentation(resource: UserResource, representation:
     }
   }
 
-  def createDeviceWithIdentityCheck(device: AddDevice, claims: Claims): Try[DeviceCreationState] = synchronized {
+  def createDeviceWithIdentityCheck(device: AddDevice, claims: Claims, userInfo: UserInfo): Try[DeviceCreationState] = synchronized {
     for {
       _ <- if (claims.hasMaybeGroups) Success(true) else Failure(InvalidClaimException("Invalid Groups", "No groups found"))
-      deviceToAdd <- Try(device.copy(listGroups = claims.targetGroups.left.map(_.map(_.toString)).merge))
+      deviceToAdd <- Try(device.copy(listGroups = claims.targetGroups.left.map(groups => {
```

Comment:
What would happend if the targetGroups are actually "right" and there are names instead of ids?

Reply:
I am not sure about the logic here, this function used 'left' before and I don't know the use case of the right at this point.

Reply:
I just remembered we always get lefts here. Otherwise, it would fail.